packages/firestore/src/core/firestore_client.ts

```diff
@@ -297,6 +300,20 @@ export class FirestoreClient {
     return true;
   }
 
+  /**
+   * Checks that the client has not been shutdown. Ensures that other methods on
+   * this class cannot be called after the client is shutdown.
+   */
+  private verifyClientRunning(): void {
+    if (this.isShutdown === false) {
+      return;
```

Comment:
Sorry, I'm not sure why I didn't notice this before, but is there a reason this isn't just:
```
if (this.isShutdown) {
  throw new FirestoreError(
      Code.FAILED_PRECONDITION,
      'The client has already been shutdown.'
    );
}
```

Reply:
my bad, that's a careless mistake from just blindly implementing changes and not considering how the rest of the code looks.